src/pages/index.js

```diff
@@ -116,25 +154,90 @@ function handlePreviewPicture(cardData) {
 }
 
 function createdCard(cardData) {
-  const card = new Card(cardData, "#card-template", handlePreviewPicture);
+  const card = new Card(
+    {
+      name: cardData.name,
+      link: cardData.link,
+      id: cardData._id,
+      isLiked: cardData.isLiked,
+    },
+    "#card-template",
+    handlePreviewPicture,
+    handleDeleteCardClick,
+    handleLikeClick
+  );
   return card.getView();
 }
 
 function handleProfileEditSubmit({ inputData }) {
-  // Update the profile using the UserInfo instance
+  // Send the profile data to the server
+  return api
+    .updateUserInfo({
+      name: inputData.name,
+      about: inputData.description,
+    })
+    .then((updatedUserInfo) => {
+      // Update the profile in the UI only after receiving a successful response
+      userInfo.setUserInfo({
+        name: updatedUserInfo.name,
+        description: updatedUserInfo.about,
+      });
+      profileModal.close();
+    })
+    .catch((error) => {
+      console.error("Error updating profile:", error);
+    });
+}
 
-  userInfo.setUserInfo({
-    name: inputData.name,
-    description: inputData.description,
+function handleAddCardSubmit({ inputData, form }) {
+  // Send the new card data to the server
+  return api
+    .addCard({
+      name: inputData.name,
+      link: inputData.link,
+    })
+    .then((newCardData) => {
+      // Render the card only after receiving a successful response from the server
+      renderCard(newCardData);
+      form.reset();
+      addCardFormValidator.toggleButtonState();
+      addCardModal.close(); // Close the add card modal
+    })
+    .catch((error) => {
+      console.error("Error adding card:", error);
+    });
+}
+
+function handleDeleteCardClick(cardId, cardInstance) {
+  deletePopup.setHandleConfirm(() => {
+    api
+      .deleteCard(cardId)
+      .then((data) => {
+        console.log(data.message); // Expected to log "This post has been deleted"
+        cardInstance.handleDeleteCard(); // Remove card from DOM
+      })
+      .catch((error) => console.error("Error deleting card:", error));
   });
 
-  profileModal.close();
+  deletePopup.open();
 }
 
-function handleAddCardSubmit({ inputData, form }) {
-  renderCard(inputData);
-  form.reset();
-  addCardFormValidator.toggleButtonState();
+function handleLikeClick(cardId, cardInstance) {
+  if (cardInstance._isLiked) {
```

Comment:
`_isLiked`

Please, delete the **low dash** (`underscore`) from methods/fields if you want to use them **outside** a class because the `low dash` shows a **private** field which is used only inside the class.